spring-amqp/src/main/java/org/springframework/amqp/core/MessagePostProcessor.java

```diff
@@ -31,6 +32,23 @@
 @FunctionalInterface
 public interface MessagePostProcessor {
 
+	/**
+	 * Change (or replace) the message.
+	 * @param message the message.
+	 * @return the message.
+	 * @throws AmqpException an exception.
+	 */
 	Message postProcessMessage(Message message) throws AmqpException;
 
+	/**
+	 * Change (or replace) the message and/or change its correlation data.
+	 * @param message the message.
+	 * @param correlation the correlation data.
+	 * @return the message.
+	 * @since 1.6.7
```

Comment:
Right; that's why my comment says I'll create a separate PR for the backport - see `CorrelationAwareMessagePostProcessor`.